In [7]:
import numpy as np
import ot
import torch
from math import sqrt
import timeit

from aux_functions import *
from TheGAN import LevyGAN
import configs_folder.configs as configs

In [10]:
actual_bsz = 65536
_s_dim = 16
total_len = actual_bsz * _s_dim

start_time = timeit.default_timer()
for i in range(100):
    result = torch.randperm(actual_bsz * _s_dim)[:actual_bsz]

elapsed = timeit.default_timer() - start_time
print(elapsed)

2.1650694419986394


In [11]:
start_time = timeit.default_timer()
for i in range(100):
    idx = 0
    result = []
    for i in range(actual_bsz):
        result.append(idx)
        idx = (idx + actual_bsz + 1) % total_len

elapsed = timeit.default_timer() - start_time
print(elapsed)

1.098054696998588


In [2]:
config = {
    'device': torch.device('cpu'),
    'ngpu': 0,
    'w dim': 4,
    'a dim': 6,
    'noise size': 62,
    'which generator': 4,
    'which discriminator': 4,
    'generator symmetry mode': 'Hsym',
    'generator last width': 6,
    's dim': 16,
    'leakyReLU slope': 0.2,
    'num epochs': 30,
    'num Chen iters': 5000,
    'optimizer': 'Adam',
    'lrG': 0.00005,
    'lrD': 0.0001,
    'beta1': 0,
    'beta2': 0.99,
    'Lipschitz mode': 'gp',
    'weight clipping limit': 0.01,
    'gp weight': 20.0,
    'batch size': 2048,
    'test batch size': 16384,
    'unfixed test batch size': 65536,
    'joint wass dist batch size': 8192,
    'num tests for 2d': 8,
    'W fixed whole': [1.0, -0.5, -1.2, -0.3, 0.7, 0.2, -0.9, 0.1, 1.7],
    'do timeing': False
}

training_config = configs.training_config

levG = LevyGAN(config)
levG.serial_number = 6
print("blub")
beta2 = 0.98
for beta1 in [0.05,0.5]:
    for msch in ['_min_sum', 'min_chen']:
        dsc = f"beta1_{beta1:.2f}_beta2_{beta2:.3f}{msch}"
        levG.load_dicts(serial_num_to_load=6, descriptor=dsc)
        print("loaded dicts")
        levG.do_tests(comp_joint_err=False, comp_grad_norm=False)
        print(f"{dsc}: chen errors: {make_pretty(levG.test_results['chen errors'])}")


blub
loaded dicts
65536
torch.Size([1048576, 10])


/home/andy/anaconda3/envs/GAN/lib/python3.10/site-packages/ot/lp/solver_1d.py:41: UserWarning: The use of `x.T` on tensors of dimension other than 2 to reverse their shape is deprecated and it will throw an error in a future release. Consider `x.mT` to transpose batches of matricesor `x.permute(*torch.arange(x.ndim - 1, -1, -1))` to reverse the dimensions of a tensor. (Triggered internally at  /opt/conda/conda-bld/pytorch_1659484746364/work/aten/src/ATen/native/TensorShape.cpp:2981.)
  cws = cws.T.contiguous()


beta1_0.05_beta2_0.980_min_sum: chen errors: ['0.4903', '0.4896', '0.2182', '0.4989', '0.2216', '0.2310']
loaded dicts
65536
torch.Size([1048576, 10])
beta1_0.05_beta2_0.980min_chen: chen errors: ['0.4428', '0.4521', '0.1645', '0.4542', '0.2168', '0.1754']
loaded dicts
65536
torch.Size([1048576, 10])
beta1_0.50_beta2_0.980_min_sum: chen errors: ['0.4950', '0.5012', '0.2236', '0.5027', '0.2285', '0.2199']
loaded dicts
65536
torch.Size([1048576, 10])
beta1_0.50_beta2_0.980min_chen: chen errors: ['0.4701', '0.4684', '0.1606', '0.4622', '0.2317', '0.1812']


beta1_0.05_beta2_0.980_min_sum:  joint err: 0.45982, chen errors: ['0.4920', '0.4912', '0.2205', '0.4960', '0.2204', '0.2306']
beta1_0.05_beta2_0.980min_chen:  joint err: 0.38241, chen errors: ['0.4405', '0.4509', '0.1649', '0.4532', '0.2158', '0.1753']
beta1_0.50_beta2_0.980_min_sum:  joint err: 0.48113, chen errors: ['0.4937', '0.5019', '0.2218', '0.4981', '0.2285', '0.2204']
beta1_0.50_beta2_0.980min_chen:  joint err: 0.38037, chen errors: ['0.4708', '0.4636', '0.1610', '0.4650', '0.2306', '0.1820']

In [2]:
print(select_pruning_indices(8,20))

[0, 1, 2, 23, 24, 25, 46, 47, 48, 69, 70, 71, 92, 93, 94, 115, 116, 117, 138, 139, 140]
